### PR TITLE
Crab netting contract - Lets approve and merge

### PR DESCRIPTION
**Pull request by sanandnarayan**
WIP PR, for @KMKoushik  to take a look

### CODE REVIEW

File: packages/crab-netting/src/CrabNetting.sol [hunk 1]
```diff
@@ -1,55 +1,992 @@
...
+        IERC20(usdc).transfer(msg.sender, _amount);
+
+        emit USDCDeQueued(msg.sender, _amount, usdBalance[msg.sender]);
+    }
+
+    /**
+     * @notice queue Crab for withdraw from crab strategy
+     * @param _amount crab amount to withdraw
+     */
+    function queueCrabForWithdrawal(uint256 _amount) external {
```
Comment (alpinechicken) [inline packages/crab-netting/src/CrabNetting.sol:337]: It's a bit confusing having 'withdraw' refer to withdrawing from the queue and withdrawing from crab. Is there better nomenclature?

Comment (alpinechicken) [reply packages/crab-netting/src/CrabNetting.sol:337]: addToQueue/removeFromQueue or queueUsdc/dequeueUscd queueCrab/dequeueCrab

File: packages/crab-netting/src/CrabNetting.sol [hunk 2]
```diff
@@ -1,55 +1,992 @@
...
+            _quantity <= IERC20(usdc).balanceOf(address(this)),
+            "Not enough deposits to net"
+        );
+        require(
+            crabQuantity <= IERC20(crab).balanceOf(address(this)),
+            "Not enough withdrawals to net"
+        );
+
+        // process deposits and send crab
+        uint256 i = depositsIndex;
```
Comment (alpinechicken) [inline packages/crab-netting/src/CrabNetting.sol:404]: Is depositsIndex initialized to zero before netAtPrice or depositAuction is called?

File: packages/crab-netting/src/CrabNetting.sol [hunk 3]
```diff
@@ -1,55 +1,992 @@
...
+     * @param _amount to deposit into crab
+     */
+    function _debtToMint(uint256 _amount) internal view returns (uint256) {
+        uint256 feeAdjustment = _calcFeeAdjustment();
+        (, , uint256 collateral, uint256 debt) = ICrabStrategyV2(crab)
+            .getVaultDetails();
+        uint256 wSqueethToMint = (_amount * debt) /
+            (collateral + (debt * feeAdjustment));
+        return wSqueethToMint;
+    }
```
Comment (alpinechicken) [inline packages/crab-netting/src/CrabNetting.sol:548]: it's so nice to be free of safemath syntax!

File: packages/crab-netting/src/CrabNetting.sol [hunk 4]
```diff
@@ -1,55 +1,996 @@
...
```
Comment (alpinechicken) [inline packages/crab-netting/src/CrabNetting.sol:548]: mms

File: packages/crab-netting/src/CrabNetting.sol (hunk 2 above)
Comment (sanandnarayan) [reply packages/crab-netting/src/CrabNetting.sol:413]: depositsIndex is a storage variable. It stores the index of the last processed Deposit Receipt

File: packages/crab-netting/src/CrabNetting.sol (hunk 4 above)
Comment (sanandnarayan) [reply packages/crab-netting/src/CrabNetting.sol:562]: didnt get you

**Comment by sanandnarayan** [comment]
This is ready for OZ

**Review by KMKoushik — APPROVED** [review]

File: packages/crab-netting/src/CrabNetting.sol (hunk 1 above)
Comment (sanandnarayan) [reply packages/crab-netting/src/CrabNetting.sol:310]: done

**Pull request merged**